plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GoogleAnalyticsApiFacade.java

```diff
@@ -0,0 +1,104 @@
+ * prohibited to anyone except those individuals and entities who have executed
+ * confidentiality and non-disclosure agreements or other agreements with Pentaho,
+ * explicitly covering such access.
+ */
+
+/* Copyright (c) 2008 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.pentaho.di.trans.steps.googleanalytics;
+
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.services.analytics.Analytics;
+import com.google.api.services.analytics.AnalyticsScopes;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import org.pentaho.di.core.util.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+
+public class GoogleAnalyticsApiFacade {
+
+  private Analytics analytics;
+
+  public static GoogleAnalyticsApiFacade createFor( String application, GaInputStepMeta meta )
+    throws IOException, GeneralSecurityException {
+
+    return new GoogleAnalyticsApiFacade(
+      GoogleNetHttpTransport.newTrustedTransport(),
+      JacksonFactory.getDefaultInstance(),
+      application,
+      meta.getOAuthServiceAccount(),
+      new File( meta.getOAuthKeyFile() ) );
+  }
+
+  private final HttpTransport httpTransport;
+
+  public GoogleAnalyticsApiFacade( HttpTransport httpTransport, JsonFactory jsonFactory, String application,
+                                   String oathServiceEmail, File keyFile ) throws IOException, GeneralSecurityException {
+    Assert.assertNotNull( httpTransport, "HttpTransport cannot be null" );
+    Assert.assertNotNull( jsonFactory, "JsonFactory cannot be null" );
+    Assert.assertNotBlank( application, "Application name cannot be empty" );
+    Assert.assertNotBlank( oathServiceEmail, "OAuth Service Email name cannot be empty" );
+    Assert.assertNotNull( keyFile, "OAuth secret key file cannot be null" );
+
+    this.httpTransport = httpTransport;
+
+    Credential credential = new GoogleCredential.Builder()
+      .setTransport( httpTransport )
+      .setJsonFactory( jsonFactory )
+      .setServiceAccountScopes( AnalyticsScopes.all() )
+      .setServiceAccountId( oathServiceEmail )
+      .setServiceAccountPrivateKeyFromP12File( keyFile )
```

Comment:
We need to put in an improvement to support:
setServiceAccountPrivateKey(PrivateKey serviceAccountPrivateKey)
and
setServiceAccountPrivateKeyFromPemFile(File pemFile)